include/graaflib/algorithm/coloring/greedy_graph_coloring.tpp

```diff
@@ -0,0 +1,45 @@
+#pragma once
+
+#include <graaflib/algorithm/coloring/greedy_graph_coloring.h>
+
+#include <unordered_map>
+
+namespace graaf::algorithm {
+
+template <typename GRAPH>
+std::unordered_map<vertex_id_t, int> greedy_graph_coloring(const GRAPH& graph) {
+  // Initialize a map to store the coloring
+  std::unordered_map<vertex_id_t, int> coloring;
+
+  // Get the vertices from the graph
+  const auto& vertices = graph.get_vertices();
+
+  // Iterate through each vertex
+  for (const auto& vertex_pair : vertices) {
+    const vertex_id_t& current_vertex_id = vertex_pair.first;
+
+    // Create a set of colors used by neighboring vertices
+    std::unordered_set<int> neighbor_colors;
```

Comment:
An example situation where this would matter:

```c++
struct foo {
  int i; // built-in type
  std::string s;  // class type
}

foo bar;  // will initialize s, but will not initialize i
foo baz{}; // initializes both s and i
```